src/tools/pbs_upgrade_job.c

```diff
@@ -90,6 +90,76 @@
 /* From pbs_ifl.h */
 #define PBS_MAXSEQNUM_PRE19	7
 #define PBS_MAXSVRJOBID_PRE19	(PBS_MAXSEQNUM_PRE19 - 1 + PBS_MAXSERVERNAME + PBS_MAXPORTNUM + 2)
+#define PBS_MAXSVRJOBID_19_20	(PBS_MAXSEQNUM - 1 + PBS_MAXSERVERNAME + PBS_MAXPORTNUM + 2)
+
+/*
+ * Replicate the jobfix structure as it was defined in 19.x and 20.x versions
+ */
+typedef struct jobfix_19_20 {
```

Comment:
Are you handling pre-19 to 19 and 19-20 upgrades all in here?

Reply:
this struct is for v19/v20 jobfix object, for pre-19, we have a different struct called jobfix_PRE19. For pre-19, I first convert it to jobfix_19_20, then convert that to the latest jobfix object.